src/plugins/event-plugins/__tests__/NavigationPlugin.test.ts

```diff
@@ -92,13 +93,43 @@ describe('NavigationPlugin tests', () => {
         // enables plugin by default
         const plugin: NavigationPlugin = buildNavigationPlugin();
 
-        // adds eventListener on load event
-        plugin.load(context);
+        // window has already loaded, so it will fire when we load.
+        // thus disabling the plugin before we load
         plugin.disable();
+        plugin.load(context);
+
         window.dispatchEvent(new Event('load'));
         plugin.disable();
 
         // Assert
         expect(record).toHaveBeenCalledTimes(0);
     });
+
+    test('when page has not loaded then we do not fire until load is completed', async () => {
+        // Setting up new mocked window that has not loaded
+        (window as any).performance = performanceEventNotLoaded.performance();
+        (window as any).PerformanceObserver =
+            performanceEvent.PerformanceObserver;
+
+        // enables plugin by default and loads
+        const plugin: NavigationPlugin = buildNavigationPlugin();
+        plugin.load(context);
+        // assert that the plugin did not fire
+        expect(record).toHaveBeenCalledTimes(0);
+
+        // now that the page has loaded, we should fire
+        window.dispatchEvent(new Event('load'));
+        expect(record).toHaveBeenCalledTimes(1);
+    });
+
+    test('when page loads before then we still fire when loading plugin', async () => {
```

Comment:
nit: 'when window.load fires before plugin loads then navigation timing is recorded'